### PR TITLE
Update CSI tests to run on k8s 1.30/1.31

### PR DESCRIPTION
**Pull request by dims**
- Drop k8s 1.27 (end of life already!)
- Add 1.30 and 1.31 k8s versions
- update hostpath_driver_version to v1.14.1

### CODE REVIEW

**Comment by xing-yang** [comment]
/hold

I'll need to check and see if anything is missing before doing this update.  Thank you @dims!

**Comment by xing-yang** [comment]
We need to get this https://github.com/kubernetes-csi/csi-driver-host-path/pull/555 merged and then cut a release for csi-driver-host-path.

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 1]
```diff
@@ -24,26 +24,28 @@ base="$(dirname $0)"
 # irrelevant because the prow.sh script will pick a suitable KinD
 # image or build from source.
 k8s_versions="
-1.27
 1.28
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/gen-jobs.sh:27]: Can you remove 1.28 too?  We usually run jobs for 3 K8s versions.

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 2]
```diff
@@ -24,26 +24,28 @@ base="$(dirname $0)"
...
 1.29
+1.30
+1.31
 "
 
 # The experimental version for which jobs are optional.
-experimental_k8s_version="1.29"
+experimental_k8s_version="1.30"
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/gen-jobs.sh:42]: I think this should be 1.31.

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 3]
```diff
@@ -24,26 +24,28 @@ base="$(dirname $0)"
...
 
 # The latest stable Kubernetes version for testing alpha jobs
-latest_stable_k8s_version="1.28"
+latest_stable_k8s_version="1.31"
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/gen-jobs.sh:45]: This should be 1.30.

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 4]
```diff
@@ -24,26 +24,28 @@ base="$(dirname $0)"
...
 
 # Tag of the hostpath driver we should use for sidecar pull jobs
-hostpath_driver_version="v1.12.1"
+hostpath_driver_version="v1.14.1"
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/gen-jobs.sh:48]: We'll cut 1.15.0 after the changelog PR is merged: https://github.com/kubernetes-csi/csi-driver-host-path/pull/555

File: config/jobs/kubernetes-csi/gen-jobs.sh (hunk 1 above)
Comment (dims) [reply config/jobs/kubernetes-csi/gen-jobs.sh:27]: on it!

File: config/jobs/kubernetes-csi/gen-jobs.sh (hunk 4 above)
Comment (dims) [reply config/jobs/kubernetes-csi/gen-jobs.sh:48]: sounds good. will update the PR, which we will be holding anyway until that happens

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 5]
```diff
@@ -24,26 +24,27 @@ base="$(dirname $0)"
...
-1.28
 1.29
+1.30
+1.31
 "
 
 # All the deployment versions we're testing.
 deployment_versions="
-1.27
 1.28
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/gen-jobs.sh:34]: Can you remove 1.28?

Comment (dims) [reply config/jobs/kubernetes-csi/gen-jobs.sh:34]: on it

**Comment by xing-yang** [comment]
csi-driver-host-path v1.15.0 is released.

/lgtm
/approve

**Pull request merged**